### PR TITLE
doc: add intro guide and a pic!

### PR DESCRIPTION
**Pull request by yuwenma**
### Change description

Fixes #

### Tests you have done

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: docs/develop-resources/README.md [hunk 1]
```diff
@@ -18,8 +22,14 @@ There are definitely more work need to be done, but we'd like to share the steps
 
 ## Develop by scenraios
```
Comment (cheftako) [inline docs/develop-resources/README.md:23]: scenraios -> scenarios

File: docs/develop-resources/README.md [hunk 2]
```diff
@@ -18,8 +22,14 @@ There are definitely more work need to be done, but we'd like to share the steps
...
 
+You may only need to add a new field, or to promote an Alpha resource to Beta rather than writing an entirely new resource. To figure out the most suitable scenario, you need to know the resources current status:
+- [ ] Check the latest [CRD](https://github.com/GoogleCloudPlatform/k8s-config-connector/tree/master/crds). If the resource exists but in Alpha, you won’t see it in  [Config Connector API Reference](https://cloud.google.com/config-connector/docs/reference/overview) until it is promoted to Beta. Please follow [promote alpha to beta](./scenarios/alpha-to-beta.md) guide.
```
Comment (cheftako) [inline docs/develop-resources/README.md:26]: These instructions are correct but assume a high level of familiarity. Maybe something more like the following?

Check the latest [CRD](https://github.com/GoogleCloudPlatform/k8s-config-connector/tree/master/crds) for the cnrm.cloud.google.com/stability-level annotation. The [Config Connector API Reference](https://cloud.google.com/config-connector/docs/reference/overview) only contains APIs of at least 'beta' stability. CRDs with 'alpha' stability need to be promoted to 'beta' stability to be formally documented in the API reference guide. Please follow the [promote alpha to beta](./scenarios/alpha-to-beta.md) guide.

File: docs/develop-resources/README.md [hunk 3]
```diff
@@ -18,8 +22,14 @@ There are definitely more work need to be done, but we'd like to share the steps
...
+- [ ] If no CRD is found,  you do need to add a new resource, please file an issue first to let us know. We can help make sure the resource is not already taken care of by someone else.
```
Comment (cheftako) [inline docs/develop-resources/README.md:27]: you do need -> you need

Comment (cheftako) [reply docs/develop-resources/README.md:27]: Also better to break this into smaller sentences.

If no CRD is found, a need resource needs to be added. To begin please see if there is already an issue filed for this resource. If there is please comment on the issue to indicate increased interest in it. If you don't find an issue for the new resource, please start by filing the issue. If you are volunteering to work on the new resource, please assign the issue to yourself. This helps make sure the resource is not already taken care of by someone else.

File: docs/develop-resources/README.md [hunk 4]
```diff
@@ -18,8 +22,14 @@ There are definitely more work need to be done, but we'd like to share the steps
...
+- [ ] We are migrating from the TF/DCL based resources to the Direct approach. That means we are holding off new PR reviews if they are using the TF/DCL based approach. Please let us know if you encounter any problems or any specific reasons that can only use the TF/DCL based approach.
```
Comment (cheftako) [inline docs/develop-resources/README.md:28]: migrating from the TF/DCL based resources -> migrating the TF/DCL based resources

Comment (cheftako) [reply docs/develop-resources/README.md:28]: That means we are holding off -> We are holding off

Comment (cheftako) [reply docs/develop-resources/README.md:28]: Please let us know if you encounter any problems or any specific reasons that can only use the TF/DCL based approach.

How? File an issue, reach out on slack, send an email, ...?

**Comment by cheftako** [comment]
Very nice improvements. A few suggestions for making the docs a bit easier to read.

File: docs/develop-resources/README.md [hunk 5]
```diff
@@ -16,10 +20,15 @@ There are definitely more work need to be done, but we'd like to share the steps
 * [4. Add Controller](./guides/4-add-controller.md)
 * [5. Releases](./guides/5-releases.md)
 
-## Develop by scenraios
+## Develop by scenarios
+
+To determine the best approach for adding support for the resource or field, please check the resource's current status:
+**Check the CRD:** Examine the latest [CRDs](https://github.com/GoogleCloudPlatform/k8s-config-connector/tree/master/crds). The version can be found from the file name. If the resource exists but is in Alpha (and therefore not yet in the [Config Connector API Reference](https://cloud.google.com/config-connector/docs/reference/overview)), follow the [Alpha to Beta promotion guide](./scenarios/alpha-to-beta.md).  This may be the simplest solution.
+**File an issue (if no CRD exists):** If no CRD is found, you'll likely need to create a new resource. Before starting this process, check if an issue is already filed. If not, please file an issue to avoid duplication of effort and allow us to coordinate effectively. If you are willing to take the resource yourself, please assign the issue.
```
Comment (yuwenma) [inline docs/develop-resources/README.md:27]: I plan to create a issue template for new resource request (to align with the direct reconciliation guide and the PM feature request guide)

**Pull request merged**